tests/expressions/tests.py

```diff
@@ -1260,6 +1260,41 @@ def test_in_lookup_allows_F_expressions_and_expressions_for_datetimes(self):
         queryset = Result.objects.filter(result_time__range=within_experiment_time)
         self.assertSequenceEqual(queryset, [r1])
 
+    def test_range_lookup_allows_F_expressions_and_expressions_for_dates(self):
```

Comment:
These tests are quite heavy but I figured style coherency was worth more than optimization here.

Reply:
I think it's fine to use `subTest()` here.

Reply:
Should you think it's worth adapting `test_in_lookup_allows_F_expressions_and_expressions_for_integers` as well to test different output fields under a single `test_in_lookup_allows_F_expressions_and_expressions` test with multiple `subTest`?

Reply:
Probably not worth changing, let's keep a data type in the test method name. Should be fine to keep datetimes and dates under `_for_dates` :thinking:

Reply:
What do you think?

Reply:
That looks great 👍

Reply:
:handshake: :tada: